src/Core/Managed/Shared/DataContracts/AvailabilityTelemetry.cs

```diff
@@ -46,6 +46,19 @@ public AvailabilityTelemetry(string name, DateTimeOffset timeStamp, TimeSpan dur
         }
 
         /// <summary>
+        /// Initializes a new instance of the <see cref="AvailabilityTelemetry"/> class by cloning an existing instance.
+        /// </summary>
+        /// <param name="source">Source instance of <see cref="AvailabilityTelemetry"/> to clone from.</param>
+        private AvailabilityTelemetry(AvailabilityTelemetry source)
+        {
+            this.Data = source.Data.DeepClone();
+            this.context = source.context.DeepClone(this.Data.properties);
+            this.Sequence = source.Sequence;
+            this.Timestamp = source.Timestamp;
+            this.Data.id = Convert.ToBase64String(BitConverter.GetBytes(WeakConcurrentRandom.Instance.Next()));
```

Comment:
After thinking about it a bit I do not think that generating new ID makes much sense. Will fix.